coderd/files/closer.go

```diff
@@ -0,0 +1,57 @@
+package files
+
+import (
+	"context"
+	"sync"
+
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+)
+
+// CacheCloser is a cache wrapper used to close all acquired files.
+// This is a more simple interface to use if opening multiple files at once.
+type CacheCloser struct {
+	cache FileAcquirer
+
+	closers []func()
+	mu      sync.Mutex
+	closed  bool
```

Comment:
we could potentially be clever and set `closers` to `nil` to represent a "closed closer", since we already need an underlying `cache` and the zero value is already invalid

```suggestion
```

Reply:
How about instead I set `cache` to `nil`.

Setting `closers` to nil still keeps `len() == 0`, which is what we usually check.